src/data/nat/digits.lean

```diff
@@ -414,3 +418,42 @@ begin
   rw of_digits_neg_one at t,
   exact t,
 end
+
+lemma digits_len_le_digits_len_succ (b n : ℕ) : (digits b n).length ≤ (digits b (n + 1)).length :=
+begin
+  cases b,
+  { -- base 0
+    cases n; simp,
+  },
+  { cases b,
+    { -- base 1
+      simp,
+    },
+    { -- base >= 2
+      apply nat.strong_induction_on n,
+      clear n,
+      intros n IH,
+      cases n,
+      { simp },
+      { repeat { rw digits_add_two_add_one },
+        by_cases hdvd : (b.succ.succ) ∣ (n.succ+1),
+        { rw nat.succ_div_of_dvd hdvd,
+          repeat { rw list.length_cons },
+          rw nat.succ_le_succ_iff,
+          apply IH,
+          exact nat.div_lt_self (by linarith) (by linarith),
+        },
+        { rw nat.succ_div_of_not_dvd hdvd,
+          refl,
+        },
+      },
+    },
+  },
+end
+
+lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length :=
```

Comment:
Can this be deduced either from some `mono` trickery or from some general lemma saying that a function `f` from `nat` to a preorder which satisfies f(n)<=f(n+1) also satisfies a<=b -> f(a)<=f(b)? [which I can't find in mathlib]

Reply:
Maybe https://leanprover-community.github.io/mathlib_docs/order/basic.html#monotone_of_monotone_nat ?

Reply:
Using `monotone_of_monotone_nat`

```lean
lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length := 
monotone_of_monotone_nat (digits_len_le_digits_len_succ b) h
```

Reply:
I forgot to ask, but should we change it to using monotone instead of induction?

Reply:
Yeah, it does make for a cleaner proof.